minard/polling.py

```diff
@@ -105,6 +105,169 @@ def polling_info(data_type, run_number):
+                    vthr = threshold[lcn] - zero_threshold[lcn]
+                    cmos_high_rates.append("%i/%i/%i: %i Hz, Vthr: %i" %\
+                            (crate, slot, channel, data_run2[lcn], vthr))
+                # Check if the channel is at high voltage
+                hv_relay_mask = relays[crate][1] << 32 | relays[crate][0]
+                if not(hv_relay_mask & (1 << (slot*4 + (3-channel//8)))):
+                    continue
+                # Check the PMT is normal or HQE
+                if types[lcn] == LOWG or \
+                   types[lcn] == NECK or \
+                   types[lcn] == FECD or \
+                   types[lcn] == BUTT or \
+                   types[lcn] == NONE:
+                    continue
+                # Check the resistor is not pulled
+                if channel_info[lcn][0]:
+                    continue
+                # For normal/HQE online PMTs, warn about channels
+                # that are fluctuating a lot. Put a low rate cut
+                # so that we don't warn about channels we don't care about 
+                if(data_run1[lcn] > 50 and data_run2[lcn] > 50):
+                    change1 = 100*((data_run2[lcn] - data_run1[lcn])/data_run1[lcn])
+                    change2 = 100*((data_run1[lcn] - data_run2[lcn])/data_run2[lcn])
+                    if change1 > pct_change or change2 > pct_change:
+                        cmos_changes.append("%i/%i/%i: %i Hz to %i Hz" %\
+                            (crate, slot, channel, data_run1[lcn], data_run2[lcn]))
+                # If the channel is not marked as low/zero/disc warn about low rates
+                if not channel_info[lcn][1]:
+                    if(data_run1[lcn] < low_rate):
+                        cmos_low_rates.append("%i/%i/%i: %i Hz" %\
+                            (crate, slot, channel, data_run1[lcn]))
+                    elif(data_run2[lcn] < low_rate):
+                        cmos_low_rates.append("%i/%i/%i: %i Hz" %\
+                            (crate, slot, channel, data_run2[lcn]))
+
+    return cmos_changes, cmos_high_rates, cmos_low_rates, run_number
+
+
+def pmt_type(conn):
+    """ Get the PMT types """
+
+    types = [0]*9728
+    sql_result = conn.execute("SELECT crate, slot, channel, type FROM pmt_info \
+                               ORDER by crate, slot, channel")
+
+    sql_result = sql_result.fetchall()
+    for crate, slot, channel, pmttype in sql_result:
+        lcn = crate*512 + slot*32 + channel
+        types[lcn] = pmttype
+
+    return types
+
+
+def channel_information(conn):
+    """ Get the channel status (whether its rpulled or occupancy issues) """
+
+    channel_info = [0]*9728
+    sql_result = conn.execute("SELECT crate,slot,channel,resistor_pulled,zero_occupancy, \
+                               low_occupancy, bad_discriminator FROM channel_status ORDER \
+                               by crate,slot,channel,timestamp ASC")
```

Comment:
Is this supposed to return the current channel status or for a specific run? If it's for a specific run you need to determine the time of the run and select the most recent channel status before that time (something like https://github.com/snoplus/minard/blob/master/minard/channeldb.py#L180). However, if you only want the current channel status you can select from the view `current_channel_status`.